tests/test_merge_chunks.py

```diff
@@ -10,23 +14,111 @@
     INCONSISTENT_INDEXED_LIST_ERROR_MESSAGE,
     cleanup_indices,
     merge,
+    merge_chat_completion_chunks,
 )
+from tests.utils.chunks import create_chunk, create_tool_call_chunk
+from tests.utils.sharing import collect_shared_mutable_objects
+
+
+class OrderConstraint(ABC):
+    @abstractmethod
+    def satisfy(self, orig_seq: Sequence[Any], seq: Sequence[Any]) -> bool:
+        pass
+
+
+@dataclass
+class Fixed(OrderConstraint):
+    idx: int
+
+    def satisfy(self, orig_seq: Sequence[Any], seq: Sequence[Any]) -> bool:
+        return orig_seq[self.idx] == seq[self.idx]
+
+
+@dataclass
+class BeforeElem(OrderConstraint):
+    elem1: Any
+    elem2: Any
+
+    def satisfy(self, orig_seq: Sequence[Any], seq: Sequence[Any]) -> bool:
+        return seq.index(self.elem1) < seq.index(self.elem2)
```

Comment:
Currently this works perfeclty fine

But I just want to note - that `seq.index(...)` doesn't find index by actual obejct id, it finds it by value

So in case someone will use this with duplicated elements, it may lead to unexpected behaviour

```ipython
In [1]: x = {'chunk': 'test'}

In [2]: y = {'chunk': 'test1'}

In [3]: z = {'chunk': 'test'}

In [4]: test = [y, x]

In [5]: test.index(x)
Out[5]: 1

In [6]: test.insert(0, z)

In [7]: test.index(x)
Out[7]: 0
```

Reply:
So let's at least leave a comment here, to state this behaviour, or rename it to something like `BeforeValue`

Reply:
Indeed, added a note and renamed the class